### PR TITLE
Add auto balance tests.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: server/cache.go [hunk 1]
```diff
@@ -408,7 +408,7 @@ func newClusterInfo(clusterRoot string) *ClusterInfo {
 	return cluster
 }
 
-func (c *ClusterInfo) addStore(store *metapb.Store) {
+func (c *ClusterInfo) addMetaStore(store *metapb.Store) {
```
Comment (shenli) [inline server/cache.go]: What's the different between Store and MetaStore?

Comment (qiuyesuifeng) [inline server/cache.go]: StoreInfo includes the status and meta store.

Comment (shenli) [inline server/cache.go]: Meta store is a type of store? Or it is the meta data of a store?

Comment (qiuyesuifeng) [inline server/cache.go]: Meta store is only for `metapb.Store`, so i use meta for this type of store, maybe both can be merged to one function.

**Comment by shenli** [comment]
LGTM

File: server/balancer.go [hunk 2]
```diff
@@ -27,8 +27,11 @@ type Balancer interface {
 
 const (
 	// If the used ratio of one storage is greater than this value,
-	// it will never be used as a selected target and it should be rebalanced.
-	maxCapacityUsedRatio = 0.4
+	// it should be rebalanced.
+	capacityUsedRatio = 0.4
```
Comment (siddontang) [inline server/balancer.go]: use minCapcityUsedRatio ?

**Comment by qiuyesuifeng** [comment]
PTAL

File: server/balancer_test.go [hunk 3]
```diff
@@ -87,6 +81,34 @@ func (s *testBalancerSuite) newClusterInfo(c *C) *ClusterInfo {
...
+		StoreId:   proto.Uint64(storeID),
+		Capacity:  proto.Uint64(capacity),
+		Available: proto.Uint64(available),
+	}
+
+	ok := clusterInfo.updateStoreStatus(stats)
+	c.Assert(ok, IsTrue)
+}
+
+func (s *testBalancerSuite) addRegionPeer(c *C, clusterInfo *ClusterInfo, storeID uint64, region *metapb.Region, leaderPeer *metapb.Peer) {
```
Comment (shenli) [inline server/balancer_test.go]: This is a test? Or an util function?

Comment (qiuyesuifeng) [inline server/balancer_test.go]: It is a helper function for test.

**Comment by siddontang** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**